Incorrect integer value exception occurs when execute `insert into tb1_cipher values(3,'c',3);`

## Bug Report

### Which version of ShardingSphere did you use?

https://github.com/apache/shardingsphere/commit/8f898cc6501afb985dc157b7d9bbe414bdfae247

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-Proxy

### Expected behavior

Execute `insert into tb1_cipher values(3,'c',3);` successfully.

### Actual behavior

```
[INFO ] 2022-07-21 14:29:38.168 [ShardingSphere-Command-2] ShardingSphere-SQL - Logic SQL: insert into tb1_cipher values(3,'c',3)
[INFO ] 2022-07-21 14:29:38.168 [ShardingSphere-Command-2] ShardingSphere-SQL - SQLStatement: MySQLInsertStatement(super=InsertStatement(super=AbstractSQLStatement(parameterCount=0, parameterMarkerSegments=[], commentSegments=[]), table=SimpleTableSegment(tableName=TableNameSegment(startIndex=12, stopIndex=21, identifier=IdentifierValue(value=tb1_cipher, quoteCharacter=NONE)), owner=Optional.empty, alias=Optional.empty), insertColumns=Optional[InsertColumnsSegment(startIndex=22, stopIndex=22, columns=[])], insertSelect=Optional.empty, values=[InsertValuesSegment(startIndex=29, stopIndex=37, values=[LiteralExpressionSegment(startIndex=30, stopIndex=30, literals=3), LiteralExpressionSegment(startIndex=32, stopIndex=34, literals=c), LiteralExpressionSegment(startIndex=36, stopIndex=36, literals=3)])]), setAssignment=Optional.empty, onDuplicateKeyColumns=Optional.empty)
[INFO ] 2022-07-21 14:29:38.168 [ShardingSphere-Command-2] ShardingSphere-SQL - Actual SQL: ds_0 ::: insert into tb1_cipher(id, aa, contact_cipher, contact) values(3, 'c', 'rPPLyyXIrNk2WYMlTPYTDQ==', 3)
[ERROR] 2022-07-21 14:29:38.253 [ShardingSphere-Command-2] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
java.sql.SQLException: Incorrect integer value: 'c' for column 'aa' at row 1
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:965)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3978)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3914)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2530)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2683)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2491)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2449)
	at com.mysql.jdbc.StatementImpl.executeInternal(StatementImpl.java:845)
	at com.mysql.jdbc.StatementImpl.execute(StatementImpl.java:939)
	at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:103)
	at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.callback.impl.ProxyStatementExecutorCallback.execute(ProxyStatementExecutorCallback.java:40)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.callback.ProxyJDBCExecutorCallback.executeSQL(ProxyJDBCExecutorCallback.java:73)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.callback.ProxyJDBCExecutorCallback.executeSQL(ProxyJDBCExecutorCallback.java:66)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.callback.ProxyJDBCExecutorCallback.executeSQL(ProxyJDBCExecutorCallback.java:44)
	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback.execute(JDBCExecutorCallback.java:89)
	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback.execute(JDBCExecutorCallback.java:69)
	at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.syncExecute(ExecutorEngine.java:135)
	at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.parallelExecute(ExecutorEngine.java:131)
	at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.execute(ExecutorEngine.java:116)
	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutor.execute(JDBCExecutor.java:65)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.executor.ProxyJDBCExecutor.execute(ProxyJDBCExecutor.java:71)
	at org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutor.useDriverToExecute(ProxySQLExecutor.java:173)
	at org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutor.execute(ProxySQLExecutor.java:133)
	at org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutor.execute(ProxySQLExecutor.java:127)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.execute(JDBCDatabaseCommunicationEngine.java:139)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.execute(JDBCDatabaseCommunicationEngine.java:71)
	at org.apache.shardingsphere.proxy.backend.text.data.impl.SchemaAssignedDatabaseBackendHandler.execute(SchemaAssignedDatabaseBackendHandler.java:63)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.execute(MySQLComQueryPacketExecutor.java:95)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:112)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:78)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

### Reason analyze (If you can)

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

Execute following sql on native database.

```sql
USE demo_ds_0;
CREATE TABLE `tb1_cipher` (
  `id` int(11) NOT NULL,
  `contact` varchar(30) DEFAULT NULL,
  `aa` int(11) DEFAULT NULL,
  `contact_cipher` varchar(60) DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
```

Then execute next sqls on ShardingSphere-Proxy:

```sql
CREATE DATABASE encrypt_db;

ADD RESOURCE ds_0 (
    URL="jdbc:mysql://localhost:3306/demo_ds_0?serverTimezone=UTC&useSSL=false",
    USER=root,
    PASSWORD=123456,
    PROPERTIES("maximumPoolSize"=50,"idleTimeout"="30000")
);

CREATE ENCRYPT RULE tb1_cipher (
    COLUMNS(
    (NAME=`contact`,PLAIN=`contact`,CIPHER=`contact_cipher`,TYPE(NAME=AES,PROPERTIES('aes-key-value'='t5714OQKZAxsmMh')))
),QUERY_WITH_CIPHER_COLUMN=FALSE);

SHOW CREATE TABLE tb1_cipher;

insert into tb1_cipher values(3,'c',3);
```

### Example codes for reproduce this issue (such as a github link).

--- a/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/merge/dal/show/EncryptShowColumnsMergedResult.java
+++ b/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/merge/dal/show/EncryptShowColumnsMergedResult.java
@@ -76,7 +76,7 @@ public abstract class EncryptShowColumnsMergedResult implements MergedResult {
             if (!encryptTable.isPresent()) {
                 return columnName;
             }
-            Optional<String> logicColumn = encryptTable.get().isCipherColumn(columnName) ? Optional.of(encryptTable.get().getLogicColumn(columnName)) : Optional.empty();
+            Optional<String> logicColumn = encryptTable.get().isCipherColumn(columnName) ? Optional.of(encryptTable.get().getLogicColumnByCipherColumn(columnName)) : Optional.empty();
             return logicColumn.orElse(columnName);
         }
         return getOriginalValue(columnIndex, type);

--- a/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/metadata/EncryptSchemaMetaDataDecorator.java
+++ b/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/metadata/EncryptSchemaMetaDataDecorator.java
@@ -28,6 +28,7 @@ import org.apache.shardingsphere.infra.metadata.database.schema.loader.model.Tab
 
 import java.util.Collection;
 import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
 import java.util.LinkedList;
 import java.util.Map;
 import java.util.Map.Entry;
@@ -56,16 +57,20 @@ public final class EncryptSchemaMetaDataDecorator implements RuleBasedSchemaMeta
     }
     
     private Collection<ColumnMetaData> getEncryptColumnMetaDataList(final EncryptTable encryptTable, final Collection<ColumnMetaData> originalColumnMetaDataList) {
-        Collection<ColumnMetaData> result = new LinkedList<>();
+        Collection<ColumnMetaData> result = new LinkedHashSet<>();
         Collection<String> plainColumns = encryptTable.getPlainColumns();
         Collection<String> assistedQueryColumns = encryptTable.getAssistedQueryColumns();
         for (ColumnMetaData each : originalColumnMetaDataList) {
             String columnName = each.getName();
-            if (encryptTable.isCipherColumn(columnName)) {
-                result.add(createColumnMetaData(encryptTable.getLogicColumn(columnName), each));
+            if (plainColumns.contains(columnName)) {
+                result.add(createColumnMetaData(encryptTable.getLogicColumnByPlainColumn(columnName), each));
                 continue;
             }
-            if (!plainColumns.contains(columnName) && !assistedQueryColumns.contains(columnName)) {
+            if (encryptTable.isCipherColumn(columnName)) {
+                result.add(createColumnMetaData(encryptTable.getLogicColumnByCipherColumn(columnName), each));
+                continue;
+            }
+            if (!assistedQueryColumns.contains(columnName)) {
                 result.add(each);
             }
         }

--- a/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rule/EncryptTable.java
+++ b/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rule/EncryptTable.java
@@ -77,18 +77,33 @@ public final class EncryptTable {
     }
     
     /**
-     * Get logic column.
+     * Get logic column by cipher column.
      * 
      * @param cipherColumn cipher column
      * @return logic column
      */
-    public String getLogicColumn(final String cipherColumn) {
+    public String getLogicColumnByCipherColumn(final String cipherColumn) {
         for (Entry<String, EncryptColumn> entry : columns.entrySet()) {
             if (entry.getValue().getCipherColumn().equals(cipherColumn)) {
                 return entry.getKey();
             }
         }
         throw new ShardingSphereException("Can not find logic column by %s.", cipherColumn);
+    }
+    
+    /**
+     * Get logic column by plain column.
+     *
+     * @param plainColumn plain column
+     * @return logic column
+     */
+    public String getLogicColumnByPlainColumn(final String plainColumn) {
+        for (Entry<String, EncryptColumn> entry : columns.entrySet()) {
+            if (entry.getValue().getPlainColumn().isPresent() && entry.getValue().getPlainColumn().get().equals(plainColumn)) {
+                return entry.getKey();
+            }
+        }
+        throw new ShardingSphereException("Can not find logic column by %s.", plainColumn);
     }
     
     /**

--- a/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/test/java/org/apache/shardingsphere/encrypt/merge/dal/show/MergedEncryptShowColumnsMergedResultTest.java
+++ b/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/test/java/org/apache/shardingsphere/encrypt/merge/dal/show/MergedEncryptShowColumnsMergedResultTest.java
@@ -76,7 +76,7 @@ public final class MergedEncryptShowColumnsMergedResultTest {
         when(result.findEncryptTable("test")).thenReturn(Optional.of(encryptTable));
         when(encryptTable.getAssistedQueryColumns()).thenReturn(Collections.singleton("assistedQuery"));
         when(encryptTable.isCipherColumn("cipher")).thenReturn(true);
-        when(encryptTable.getLogicColumn("cipher")).thenReturn("id");
+        when(encryptTable.getLogicColumnByCipherColumn("cipher")).thenReturn("id");
         return result;
     }
     

--- a/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/test/java/org/apache/shardingsphere/encrypt/metadata/EncryptSchemaMetaDataDecoratorTest.java
+++ b/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/test/java/org/apache/shardingsphere/encrypt/metadata/EncryptSchemaMetaDataDecoratorTest.java
@@ -64,7 +64,8 @@ public final class EncryptSchemaMetaDataDecoratorTest {
         when(encryptTable.getAssistedQueryColumns()).thenReturn(Collections.emptyList());
         when(encryptTable.getPlainColumns()).thenReturn(Collections.singleton("pwd_plain"));
         when(encryptTable.isCipherColumn("pwd_cipher")).thenReturn(true);
-        when(encryptTable.getLogicColumn("pwd_cipher")).thenReturn("pwd");
+        when(encryptTable.getLogicColumnByCipherColumn("pwd_cipher")).thenReturn("pwd");
+        when(encryptTable.getLogicColumnByPlainColumn("pwd_plain")).thenReturn("pwd");
         return result;
     }
     

--- a/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/test/java/org/apache/shardingsphere/encrypt/rule/EncryptTableTest.java
+++ b/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/test/java/org/apache/shardingsphere/encrypt/rule/EncryptTableTest.java
@@ -58,13 +58,23 @@ public final class EncryptTableTest {
     }
     
     @Test
-    public void assertGetLogicColumn() {
-        assertNotNull(encryptTable.getLogicColumn("cipherColumn"));
+    public void assertGetLogicColumnByCipherColumn() {
+        assertNotNull(encryptTable.getLogicColumnByCipherColumn("cipherColumn"));
     }
     
     @Test(expected = ShardingSphereException.class)
-    public void assertGetLogicColumnWhenNotFind() {
-        encryptTable.getLogicColumn("invalidColumn");
+    public void assertGetLogicColumnByCipherColumnWhenNotFind() {
+        encryptTable.getLogicColumnByCipherColumn("invalidColumn");
+    }
+    
+    @Test
+    public void assertGetLogicColumnByPlainColumn() {
+        assertNotNull(encryptTable.getLogicColumnByPlainColumn("plainColumn"));
+    }
+    
+    @Test(expected = ShardingSphereException.class)
+    public void assertGetLogicColumnByPlainColumnWhenNotFind() {
+        encryptTable.getLogicColumnByPlainColumn("invalidColumn");
     }
     
     @Test

--- a/shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-boot-starter/src/test/java/org/apache/shardingsphere/spring/boot/SpringBootStarterTest.java
+++ b/shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-boot-starter/src/test/java/org/apache/shardingsphere/spring/boot/SpringBootStarterTest.java
@@ -157,7 +157,7 @@ public class SpringBootStarterTest {
     }
     
     private void assertEncryptTable(final EncryptTable actual) {
-        assertThat(actual.getLogicColumn("pwd_cipher"), is("pwd"));
+        assertThat(actual.getLogicColumnByCipherColumn("pwd_cipher"), is("pwd"));
         assertThat(actual.getPlainColumns(), is(Collections.singletonList("pwd_plain")));
         assertThat(actual.getAssistedQueryColumns(), is(Collections.singletonList("pwd_assisted_query_cipher")));
     }

